src/plugins/vis_augmenter/public/utils/utils.test.ts

```diff
@@ -16,26 +16,150 @@ import {
+      typesRegistry,
+    });
+    const validVis = ({
+      params: {
+        type: 'line',
+        seriesParams: [
+          {
+            type: 'line',
+          },
+        ],
+      },
+      data: {
+        aggs,
+      },
+    } as unknown) as Vis;
+    it('vis is ineligible with invalid non-line type', async () => {
       const vis = ({
         params: {
           type: 'not-line',
+          seriesParams: [],
+        },
+        data: {
+          aggs,
         },
       } as unknown) as Vis;
-      expect(isEligibleForVisLayers(vis)).toEqual(false);
+      expect(isEligibleForVisLayers(vis, validDimensions)).toEqual(false);
     });
-    it('vis is eligible with valid type', async () => {
+    it('vis is ineligible with no date_histogram', async () => {
+      const invalidConfigStates = [
+        {
+          enabled: true,
+          type: 'histogram',
+          params: {},
+        },
+        {
+          enabled: true,
+          type: 'metrics',
+          params: {},
+        },
+      ];
+      const invalidAggs = new AggConfigs(
+        stubIndexPatternWithFields as IndexPattern,
+        invalidConfigStates,
+        {
+          typesRegistry,
+        }
+      );
+      const vis = ({
+        params: {
+          type: 'line',
+          seriesParams: [],
+        },
+        data: {
+          invalidAggs,
+        },
+      } as unknown) as Vis;
+      expect(isEligibleForVisLayers(vis, validDimensions)).toEqual(false);
+    });
+    it('vis is ineligible with invalid aggs counts', async () => {
+      const invalidConfigStates = [
+        {
+          enabled: true,
+          type: 'date_histogram',
+          params: {},
+        },
+        {
+          enabled: true,
+          type: 'date_histogram',
+          params: {},
+        },
+        {
+          enabled: true,
+          type: 'metrics',
+          params: {},
+        },
+      ];
+      const invalidAggs = new AggConfigs(
+        stubIndexPatternWithFields as IndexPattern,
+        invalidConfigStates,
+        {
+          typesRegistry,
+        }
+      );
+      const vis = ({
+        params: {
+          type: 'line',
+          seriesParams: [],
+        },
+        data: {
+          invalidAggs,
+        },
+      } as unknown) as Vis;
+      expect(isEligibleForVisLayers(vis, validDimensions)).toEqual(false);
+    });
+    it('vis is ineligible with series param is not line type', async () => {
       const vis = ({
         params: {
           type: 'line',
+          seriesParams: [
+            {
+              type: 'area',
+            },
+          ],
+        },
+        data: {
+          aggs,
         },
       } as unknown) as Vis;
-      expect(isEligibleForVisLayers(vis)).toEqual(true);
+      expect(isEligibleForVisLayers(vis, validDimensions)).toEqual(false);
+    });
+    it('vis is eligible with valid type', async () => {
+      expect(isEligibleForVisLayers(validVis, validDimensions)).toEqual(true);
```

Comment:
Can you add a test for invalid dimensions?

Reply:
done